apps/labs/posts/intel-oneapi-msvc-spin-scipy.md

```diff
@@ -0,0 +1,174 @@
+This means improvements made to `spin` will automatically benefit SciPy.
+Additionally, any limitations we encounter in `spin` will be addressed through Pull Requests, making it a two-way connection: SciPy benefits from other projects (via `spin`), and those projects benefit from improvements made by SciPy.
+
+Let me share three significant examples of this two-way connection,
+
+**Passing Arguments to `meson compile` and `meson install`**
+
+Initially, `spin` didn’t support passing arguments to `meson compile` or `meson install`.
+It only forwarded arguments to the `meson setup` step. Since SciPy requires passing certain arguments (like `--tags`) to `meson install` [[gh-20712](https://github.com/scipy/scipy/pull/20712)], we needed a way to forward those arguments through `spin`.
+I contributed a PR to `spin` to allow for CLI arguments to be passed to `meson compile` and `meson install` steps [[scientific-python/spin#256]](https://github.com/scientific-python/spin/pull/256), which was merged.
+This enhancement allows SciPy to seamlessly pass the necessary arguments, further improving its integration with `spin`.
+
+**Reducing Code Complexity**
+
+`spin` allowed us to simplify the implementation of `dev.py` functionality.
+It offers a decorator, `spin.util.extend_command`, which enables the extension of `spin` commands (e.g., `spin build`, `spin test`) within a user project (in our case, SciPy).
+By using this decorator, we implemented SciPy-specific checks while leveraging `spin`'s core functionality.
+As a result, the code size was reduced by 44% compared to `dev.py`.
+This reduction in code complexity means a smaller surface area for bugs — another strong reason for adopting `spin`.
+
+**Fixing a linker path issue on macOS**
+
+While building SciPy with `spin`, we encountered an issue on macOS where the linker path for shared libraries became corrupted due to the `--destdir` specification in the `meson install` step.
+This issue arose because `spin` sets `/usr` or `C:\` as the default value for `--prefix`, but it also used `--destdir` during installation.
+`--destdir` is typically used for packaging, not for final installations.
+To resolve this, we removed the `--destdir` usage and set the `--prefix` directly to the installation directory.
+The fix was submitted as a PR to `spin` [[scientific-python/spin#257]](https://github.com/scientific-python/spin/pull/257), ensuring smoother installation behavior for SciPy on macOS.
+
+### Current Status of `spin` in SciPy
+
+We are close to merging `DEV: use spin (prototype)` [[gh-21674](https://github.com/scipy/scipy/pull/21674)].
+We’re waiting for the PR [scientific-python/spin#257](https://github.com/scientific-python/spin/pull/257) to be merged, after which we’ll be ready to fully integrate `spin` into SciPy.
+Our work has been fully tested in the SciPy CI system, ensuring its stability.
+
+Here’s a quick comparison of the SciPy build experience with `dev.py` versus `spin`:
```

Comment:
The videos don't seem to rendering in the preview: https://labs-git-fork-czgdp1807-feature-intel-oneapi-spin-quansight.vercel.app/blog/intel-oneapi-msvc-spin-scipy

Reply:
I see, I can try https://docs.asciinema.org/manual/server/embedding/#preview-image-link for sharing the videos. `asciinema` will provide an image. If someone clicks it then the videos plays on asciinema website. That should solve the problems showing up due to embedding the player using `script` tag. Let me know if it would work.

Reply:
That seems like a pragmatic solution - try it I'd say.

Reply:
I pushed up a commit to do this (please let me know if you're not comfortable with this, and I'll keep it mind for the future), and unblock merging this PR. :)

Reply:
No worries. Thanks for doing it. :D.